.github/workflows/ruby.yml

```diff
@@ -7,17 +7,46 @@ on:
     branches: [ master ]
 
 jobs:
-  test:
+  test_matrix:
 
     runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby:
+          - 2.4.10
+          - 2.5.8
+          - 2.6.6
+          - 2.7.2
+          - 3.0.0
+
+        rails:
+          - '52'
+          - '60'
+
+        exclude:
+          - ruby: 2.4.10
+            rails: '60'
+          - ruby: 3.0.0
+            rails: '52'
+
+    env:
+      BUNDLE_GEMFILE: gemfiles/rails_${{ matrix.rails }}.gemfile
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
```

Comment:
I see we're using bundler-cache instead of the bundle install step, what changes does this do to the flow/how does it handle the caching?

Reply:
This runs `bundle install` in the scope of specified Gemfile, and caches the result with `actions/cache`

Reply:
This basically handles 3 steps:
- setup ruby
- restore/persist cache (considering Gemfile lock, ruby and platform version asking the way)
- run bundle install

Reply:
Thanks!